### PR TITLE
fix ZEN-21760

### PR DESCRIPTION
**Pull request by ssaporylo**
https://jira.zenoss.com/browse/ZEN-21760

### CODE REVIEW

File: ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py [hunk 1]
```diff
@@ -155,16 +156,24 @@ def create_proxy_device(self):
 
         # add the missing proxy device.
         device_name = self.name()
+        device = self.dmd.Devices.findDeviceByIdOrIp(getHostByName(device_name))
         if self.dmd.Devices.findDevice(device_name):
```
Comment (ssvsergeyev) [inline ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py]: Does it not the same as previous line with getting device by device_name?

File: ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py [hunk 2]
```diff
@@ -155,16 +156,24 @@ def create_proxy_device(self):
...
             device_name = device_name + "_nameconflict"
-            LOG.info("Device name conflict with endpoint.  Changed name to %s" % device_name)
+            LOG.info("Device name conflict with endpoint.  Changed name to %s"
+                     % device_name)
 
-        LOG.info('Adding device for %s %s' % (self.meta_type, self.title))
+        if device:
```
Comment (ssvsergeyev) [inline ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py]: Maybe it same as on line with prev comment?

File: ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py (hunk 1 above)
Comment (ssaporylo) [inline ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py]: On line 159 I want to get device by Ip

File: ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py [hunk 3]
```diff
@@ -155,16 +156,28 @@ def create_proxy_device(self):
...
         # add the missing proxy device.
         device_name = self.name()
+        try:
+            device = self.dmd.Devices.findDeviceByIdOrIp(
+                getHostByName(device_name))
+        except Exception:
+            device = None
         if self.dmd.Devices.findDevice(device_name):
             device_name = device_name + "_nameconflict"
-            LOG.info("Device name conflict with endpoint.  Changed name to %s" % device_name)
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/DeviceProxyComponent.py]: Don't change formatting unless content changes.

**Comment by pcarinhas** [comment]
* Traceback during testing: http://paste.zenoss.loc/view/a569878e
* Created a device with a short non-dns name and manageIp. (kilo)
* Created osi212 OSI device. During modeling phase we get traceback.

**Pull request merged**